drivers/dcf77/dcf77.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup     drivers_dcf77
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the dcf 77
+ *              longwave time signal and standard-frequency radio station
+ *
+ * @author      Michel Gerlach <michel.gerlach@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include <string.h>
+
+#include "log.h"
+#include "assert.h"
+#include "xtimer.h"
+#include "timex.h"
+#include "periph/gpio.h"
+
+#include "dcf77.h"
+#include "dcf77_internal.h"
+#include "dcf77_params.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/* Persistent level longer than 1200ms starts a new cycle  */
+#define DCF77_PULSE_START_HIGH_THRESHOLD        (1200000U)  /*~1000ms*/
+/* Every pulse send by the DCF longer than 130ms is interpreted as 1 */
+#define DCF77_PULSE_WIDTH_THRESHOLD             (140000U)   /*~130ms*/
+/* If an expected pulse is not detected within 2,5s, something is wrong */
+#define DCF77_TIMEOUT                           (2500000U)  /*~2500ms*/
+/* Number of bits in a cycle*/
+#define DCF77_READING_CYCLE                     (59)
+
+#define DCF77_MINUTE_MASK                       (0xFE00000ULL)
+#define DCF77_HOUR_MASK                         (0x7E0000000ULL)
+#define DCF77_DATE_MASK                         (0x1FFFFFC00000000ULL)
+
+#define DCF77_MINUTE_SHIFT                      (21)
+#define DCF77_HOUR_SHIFT                        (29)
+#define DCF77_DATE_SHIFT                        (36)
+
+static void _level_cb_high(dcf77_t *dev)
+{
+    switch (dev->internal_state) {
+        case DCF77_STATE_START:
+            DEBUG("[dcf77] EVENT START 1 !\n");
+            dev->stopTime = xtimer_now_usec();
+            if ((dev->stopTime - dev->startTime) >
```

Comment:
```suggestion
            if ((xtimer_now_usec() - dev->startTime) >
```

`stopTime` is not used anywhere else.